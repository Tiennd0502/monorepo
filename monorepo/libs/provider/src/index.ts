export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
