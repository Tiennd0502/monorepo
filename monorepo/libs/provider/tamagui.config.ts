import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
