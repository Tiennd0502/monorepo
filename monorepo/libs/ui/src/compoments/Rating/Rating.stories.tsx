import { Meta, StoryObj } from '@storybook/react';

import Rating from '.';

const meta: Meta<typeof Rating> = {
  title: 'Components/Rating',
  component: Rating,
};

export default meta;

type Story = StoryObj<typeof Rating>;

export const Default: Story = {
  args: {
    value: 3,
  },
};

export const HalfGrad: Story = {
  args: {
    value: 3.5,
    size: 20,
  },
};
