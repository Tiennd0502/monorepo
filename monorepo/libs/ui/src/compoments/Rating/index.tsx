import { memo } from 'react';

import { StarIcon } from '../icons';
import { XStack } from 'tamagui';

interface RatingProps {
  value: number;
  size?: number | string;
  count?: number;
}

const Rating = ({ value, size, count = 5 }: RatingProps) => {
  return (
    <XStack gap="$1">
      {[...Array(count).keys()].map((item) => (
        <StarIcon
          key={item}
          width={size}
          height={size}
          isActive={item < value}
          isHalfGrad={item > value - 1}
        />
      ))}
    </XStack>
  );
};

export default memo(Rating);
