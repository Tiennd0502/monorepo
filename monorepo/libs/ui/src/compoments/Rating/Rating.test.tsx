import {render} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import Rating from '.';

const baseProps = {
  value: 5
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <Rating {...props} />
    </TamaguiProvider>
  );
};

describe('Rating', () => {
  it('should render Rating component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render Rating component with value is a decimal number', () => {
    const container = setup({value: 2.5, size: 20});

    expect(container).toMatchSnapshot();
  });
});
