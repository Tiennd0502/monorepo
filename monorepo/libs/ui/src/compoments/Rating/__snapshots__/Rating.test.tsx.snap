// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render Rating component 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "gap": 4,
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    fill="none"
    focusable={false}
    height="16"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="16"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -865972,
              0.5,
              -865972,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    fill="none"
    focusable={false}
    height="16"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="16"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -865972,
              0.5,
              -865972,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    fill="none"
    focusable={false}
    height="16"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="16"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -865972,
              0.5,
              -865972,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    fill="none"
    focusable={false}
    height="16"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="16"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -865972,
              0.5,
              -865972,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    fill="none"
    focusable={false}
    height="16"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="16"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -865972,
              0.5,
              -865972,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Rating should render Rating component with value is a decimal number 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "gap": 4,
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="20"
    bbWidth="20"
    fill="none"
    focusable={false}
    height="20"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="20"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -865972,
              0.5,
              -865972,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="20"
    bbWidth="20"
    fill="none"
    focusable={false}
    height="20"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="20"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -865972,
              0.5,
              -865972,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="20"
    bbWidth="20"
    fill="none"
    focusable={false}
    height="20"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="20"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -865972,
              0.5,
              -6710887,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="20"
    bbWidth="20"
    fill="none"
    focusable={false}
    height="20"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="20"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -6710887,
              0.5,
              -6710887,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="20"
    bbWidth="20"
    fill="none"
    focusable={false}
    height="20"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width="20"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            [
              0.5,
              -6710887,
              0.5,
              -6710887,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="a"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        />
      </RNSVGDefs>
      <RNSVGPath
        d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
        fill={
          {
            "brushRef": "a",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
